ddtrace/tracer.py

```diff
@@ -633,7 +667,13 @@ def set_service_info(self, *args, **kwargs):
         """Set the information about the given service."""
         return
 
-    def wrap(self, name=None, service=None, resource=None, span_type=None):
+    def wrap(
+        self,
+        name=None,  # type: Optional[str]
+        service=None,  # type: Optional[str]
+        resource=None,  # type: Optional[str]
+        span_type=None,  # type: Optional[str]
```

Comment:
Is this better written as:

```python
def wrap(...):
    # type: (Optional[str], Optional[str], Optional[str], Optional[str]) -> Callable[ <> ]
```

This way we can get the return type as well?

Reply:
nvm: https://github.com/DataDog/dd-trace-py/pull/2166#discussion_r594545609

a return type would be nice though here and on `def trace()`

Reply:
You can add the return type below with
```python
# type: (...) -> Callable
```